tensorflow_model_optimization/python/core/sparsity_tf2/schedule.py

```diff
@@ -235,32 +278,54 @@ def __call__(self, step):
     return (should_prune_in_step,
         self._get_update_percentage(self.alpha, should_prune_in_step, step))
 
+
 class LRSchedule(Schedule):
   """Scales the drop fraction according to the learning rate."""
   def __init__(self,
                initial_drop_fraction,
                begin_step,
-               optimizer,
-               initial_lr=0.01,
                end_step=-1,
-               frequency=100):
+               frequency=100,
+               optimizer=None):
     """Initializes an annealing schedule for the drop fraction in a
     mask update function that follows the learning rate.
-
     Args:
       optimizer: the optimizer being used at training time
```

Comment:
We talked about this offline: this should take a learning rate schedule instead of an optimizer to prevent subtle serialization/deserialization bugs in practice